assets/proto/element.proto

```diff
@@ -0,0 +1,141 @@
+syntax = "proto2";
+package web_stories_editor;
+
+import "resource.proto";
+import "types.proto";
+
+// An element on a story page.
+message Element {
+  // Element ID.
+  required string id = 1;
+
+  // Element type.
+  required ElementType type = 2;
+
+  // Opacity: 0-100%.
+  required int32 opacity = 3 [default = 100];
+
+  // Box: {x, y, width, height} in px.
+  // The px coordinate system is 440x660.
+  required int32 x = 4;
+  required int32 y = 5;
+  required int32 width = 6;
+  required int32 height = 7;
+
+  // Rotation angle: 0-360deg.
+  required int32 rotation_angle = 8 [default = 0];
+
+  // Overrides x/y/width/height/rotation_angle and makes an element to take
+  // the complete "safe area" of the page.
+  required bool is_fill = 9 [default = false];
+
+  // The vertical/horizontal flip.
+  // Default: "no flip".
+  optional Flip flip = 10;
+
+  // The clipping mask of the element.
+  // Default: "no mask" or a simple rectangle mask.
+  optional Mask mask = 11;
+
+  // TODO: not sure if an extension or just optional fields are better here.
+  extensions 1000 to max;
```

Comment:
I like extensions

Reply:
Am I using them correctly here? E.g. would I be able to do `elements = [new MediaElement(), new TextElement()]`, etc on the `repeated Element elements`? Also, it bugs me:

1. I can't have an extension w/o fields. E.g. I also want to have `ImageElement extends MediaElement`  that, for now, might not have any fields.
2. There are two things responsible for type: the field "type" and the type-base message, e.g. `TextElement`. Is there a better way to tie them together?